### PR TITLE
add trunk-ng, a fork of trunk

### PR DESCRIPTION
**Pull request by ctron**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by v-kat** [comment]
Thanks a lot for doing this!

**Comment by ctron** [comment]
Is there something I need to do?

**Comment by ctron** [comment]
@NickCao can you help?

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3701,6 +3701,12 @@
     githubId = 490965;
     name = "Craig Swank";
   };
+  ctron = {
+    email = "ctron@dentrassi.de";
+    github = "ctron";
+    githubId = 202474;
+    name = "Jens Reimann";
+  };
```
Comment (NickCao) [inline maintainers/maintainer-list.nix:3709]: Please split the maintainer addition to its own commit.

File: pkgs/development/tools/trunk-ng/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "trunk-ng";
+  version = "0.17.8";
+
+  src = fetchFromGitHub {
+    owner = "ctron";
+    repo = "trunk";
+    rev = "v${version}";
+    sha256 = "sha256-ycZIqDBZccPapOK0ZI9Cvq94tRxChrsWX1rhyWh0S2c=";
```
Comment (NickCao) [inline pkgs/development/tools/trunk-ng/default.nix:12]:
```suggestion
    hash = "sha256-ycZIqDBZccPapOK0ZI9Cvq94tRxChrsWX1rhyWh0S2c=";
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -14063,6 +14063,10 @@ with pkgs;
 
   trunk-io = callPackage ../development/tools/trunk-io { };
 
+  trunk-ng = callPackage ../development/tools/trunk-ng {
+    inherit (darwin.apple_sdk.frameworks) CoreServices Security;
+  };
```
Comment (NickCao) [inline pkgs/top-level/all-packages.nix:14068]: Try moving it to the new `by-name` hierarchy, see https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name

**Comment by ctron** [comment]
@NickCao thanks for taking the time, I think I addressed the comments. Split up the two commits and applied the changes.

**Comment by ctron** [comment]
> Diff LGTM, only need to fix the commit messages. (See https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)

Changed those.

**Review by NickCao — APPROVED** [review]
Just waiting for ofborg.
